### ISSUE TITLE
invalid argument: could not find capabilities for domaintype=kvm

### ISSUE THREAD

**Issue by lifeofguenter** [dmacvicar/terraform-provider-libvirt#442]
#  Version Reports:

### Distro  version of host:

``` Debian 9 ```

### Terraform Version Report

```Terraform v0.11.8```

### Libvirt version

```4.8.0```

### terraform-provider-libvirt plugin version (git-hash)

``` commit 8679775a0117011fbc47b9f3b330015e7779ee29 (HEAD, tag: v0.4.4)```
___
# Description of Issue/Question

### Setup

```
resource "libvirt_domain" "jenkinsn" {
  name      = "jenkinsn"
  autostart = true

  cpu {
    mode = "host-passthrough"
  }
  vcpu    = 1
  memory  = 256

  disk {
    volume_id = "${libvirt_volume.jenkinsn_vol_root.id}"
    scsi      = true
  }
}

resource "libvirt_volume" "jenkinsn_vol_root" {
  name   = "jenkinsn_vol_root"
  size   = "${1 * 1024 * 1024 * 1024}"
  format = "qcow2"
}
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

___
# Additional Infos:

I am running libvirt with qemu (without kvm) inside a VirtualBox VM. According to https://libvirt.org/drvqemu.html#prereq that should be perfectly fine as the host will check against `/usr/bin/kvm` for kvm capabilities.

However I am getting the following error:

```
* libvirt_domain.jenkinsn: Error defining libvirt domain: virError(Code=8, Domain=44, Message='invalid argument: could not find capabilities for domaintype=kvm ')
```

It would be great if one could set the `type` as specified here: https://libvirt.org/formatdomain.html#elements

**Comment by MalloZup** [comment]
https://github.com/dmacvicar/terraform-provider-libvirt/pull/431

**Comment by MalloZup** [comment]
for this custom feature we will take care soon on that PR

**Comment by bzub** [comment]
Hi @lifeofguenter, as @MalloZup mentioned you can change the domain using the xml/xslt feature.

Here's an example that worked for me to use qemu instead of kvm:
```terraform
resource "libvirt_domain" "terraform_test" {
  name     = "terraform_test"

  xml {
    xslt = <<EOF
<?xml version="1.0" ?>
<xsl:stylesheet version="1.0"
                xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
  <xsl:output omit-xml-declaration="yes" indent="yes"/>
  <xsl:template match="node()|@*">
     <xsl:copy>
       <xsl:apply-templates select="node()|@*"/>
     </xsl:copy>
  </xsl:template>

	<xsl:template match="@type[. = 'kvm']">
		<xsl:attribute name="type">
			<xsl:value-of select="'qemu'"/>
		</xsl:attribute>
	</xsl:template>
</xsl:stylesheet>
EOF
  }
}
```

I believe this issue could be closed?

**Comment by MalloZup** [comment]
True. Thx..!

**Comment by MalloZup** [comment]
Feel free to ask any add info here

**Comment by qops1981** [comment]
I am not sure if I have the same scenario or not. 
I am operating from a Fedora VM where I have also installed libvirt to build this plugins project.
I connect to a remote libvirt using `virsh -c "qemu+ssh://<remote system>/system"` and all that works fine.
When I try to use terraform to manage that remote libvirt system I get that error.
The remote system is newer and the versions of libvirt etc are newer

```
.terraform.d/plugins/terraform-provider-libvirt 6828269db97e943dad81219a41a16091b426f954
Compiled against library: libvirt 3.2.1
Using library: libvirt 3.2.1
Running hypervisor: QEMU 2.9.1
Running against daemon: 3.2.1
```
```
virsh -c qemu:///system version --daemon
Compiled against library: libvirt 4.0.0
Using library: libvirt 4.0.0
Using API: QEMU 4.0.0
Running hypervisor: QEMU 2.11.1
Running against daemon: 4.0.0
```

I hope there isn't an incompatibility there. 
I have tried the above XML block and that did do the trick. I see though some time has past. Is that still the recommended solution? Or has there been a feature added where I can just call out some parameter in the `libvirt_domain` block? Either way I am fine, just trying to stay canonical to the plugin as it stands.

Thank you for your time.
